libs/platform/experience/entity-text/entity-text.component.ts

```diff
@@ -33,14 +32,18 @@ export class EntityTextComponent extends TextMixin(
   });
 
   protected $link = computed(() => {
-    const { entity, field } = this.$options();
-    if (!entity || !field) return;
+    const { entity: type } = this.$options();
 
-    const qualifier = this.contextService.get(this, field);
-    return this.linkService.get({
-      type: entity,
-      qualifier,
-    });
+    return this.entityService.getQualifier({ element: this, type }).pipe(
+      switchMap(({ type, qualifier }) => {
+        console.log(qualifier, type);
+        return this.linkService.get({
+          type,
+          qualifier,
+        });
+      }),
+      catchError(() => of())
```

Comment:
Can we move error handling to the service?

Reply:
we can, but I'm not sure we should do it in general, as it means, we are swallowing errors inside service, and we can't implement more complex handling for some specific cases outside of it

Reply:
ok, then should we know exact what we are handling? Or are we handling any error? For the current case entity service can throw an error that means that some data is wrong and this is importatn, why should we catch it inside component then?

Reply:
In this component we just don't display anything if entity resolving is invalid (or not show the link) and that's the desired behavior here.